include/natalie/symbol_object.hpp

```diff
@@ -68,21 +69,32 @@ class SymbolObject : public Object {
 
     SymbolObject(Env *env, const char *name)
         : Object { Object::Type::Symbol, GlobalEnv::the()->Symbol() }
+        , m_should_free { true }
         , m_name { strdup(name) } {
         assert(m_name);
     }
 
     SymbolObject(const char *name)
         : Object { Object::Type::Symbol, GlobalEnv::the()->Symbol() }
+        , m_should_free { true }
         , m_name { strdup(name) } {
         assert(m_name);
     }
+    // FIXME: find a better semantic for this,
+    //        maybe some sort of badge
+    SymbolObject(const char *name, size_t)
+        : Object { Object::Type::Symbol, GlobalEnv::the()->Symbol() }
+        , m_should_free { false }
+        , m_name { name } {
+        assert(m_name);
+    }
 
+    bool m_should_free { false };
```

Comment:
Let's go with something like this:

```suggestion
    bool m_owned_string { false };
```